cmake/CMakeLists.txt

```diff
@@ -603,6 +603,18 @@ if(PKG_OPT)
 endif()
 
 if(PKG_USER-INTEL)
+    if (NOT CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
+      message(FATAL_ERROR "USER-INTEL is only useful together with intel compiler")
+    endif()
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "2016")
+      message(FATAL_ERROR "USER-INTEL is needed at least 2016 intel compiler")
+    endif()
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "2017.3")
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -xHost")
+    else()
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -xCOMMON-AVX512")
+    endif()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -qopenmp -qno-offload -fno-alias -ansi-alias -restrict -DLMP_INTEL_USELRT -DLMP_USE_MKL_RNG-O2 -fp-model fast=2 -no-prec-div -qoverride-limits -qopt-zmm-usage=high") 
```

Comment:
@wmbrownIntel : are these the right flags `-qopenmp -qno-offload -fno-alias -ansi-alias -restrict -DLMP_INTEL_USELRT -DLMP_USE_MKL_RNG-O2 -fp-model fast=2 -no-prec-div -qoverride-limits -qopt-zmm-usage=high` plus `-xHost` (or -`xCOMMON-AVX512` for >-icc-2017.3)?